compiler/rustc_trait_selection/src/traits/fulfill.rs

```diff
@@ -378,9 +383,9 @@ impl<'a, 'b, 'tcx> FulfillProcessor<'a, 'b, 'tcx> {
                 | ty::PredicateAtom::ConstEvaluatable(..)
                 | ty::PredicateAtom::ConstEquate(..) => {
                     let pred = infcx.replace_bound_vars_with_placeholders(binder);
-                    ProcessResult::Changed(mk_pending(vec![
-                        obligation.with(pred.to_predicate(self.selcx.tcx())),
-                    ]))
+                    let v: SmallVec<[_; 1]> =
+                        smallvec![obligation.with(pred.to_predicate(self.selcx.tcx()))];
+                    ProcessResult::Changed(mk_pending(v))
```

Comment:
```suggestion
                    ProcessResult::Changed(mk_pending(
                      [obligation.with(pred.to_predicate(self.selcx.tcx()))].iter().copied()
                    ))
```
I also think that using a `vec` here is fine, I don't think that this code is hot enough for this to matter and adding a generic parameter to `mk_pending` probably increases the emitted llvmir which does not seem like a positive trade to me.

Reply:
I'll see what can be done here without making `mk_pending` generic.